File: WeeklyAssignments/week5/bitAnd.java
public class bitAnd {
    public static void main(String[] args) {
        int a = 6;
        int b = 5;
        System.out.println("Answer with & operator: " + (a&b));
        // ~(a&b) = ~a | ~b
        // ~(~(a&b)) = ~(~a | ~b)
        int ans = ~(~a | ~b);
        System.out.println("Answer without & operator: " + ans);
    }
}

File: WeeklyAssignments/week5/bitAnd.c
#include <stdio.h>
int bitAndOp(int x, int y) {
    // ~(x&y) = ~x | ~y
    // ~(~(x&y)) = ~(~x | ~y)
    return ~(~x | ~y);
}

int main(){
    int a = 6;
    int b = 5;
    printf("Answer with & operator:%d\n", (a&b));
    printf("Answer without & operator:%d\n", bitAndOp(a, b));
    return 0;
}


